be/src/formats/parquet/meta_helper.cpp

```diff
@@ -23,30 +23,64 @@
 namespace starrocks::parquet {
 
 void ParquetMetaHelper::build_column_name_2_pos_in_meta(
-        std::unordered_map<std::string, size_t>& column_name_2_pos_in_meta, const tparquet::RowGroup& row_group,
+        std::unordered_map<std::string, size_t>& column_name_2_pos_in_meta,
         const std::vector<SlotDescriptor*>& slots) const {
+    auto& schema = _file_metadata->schema();
     for (const auto& slot : slots) {
-        const std::string format_slot_name = Utils::format_name(slot->col_name(), _case_sensitive);
-        for (size_t idx = 0; idx < row_group.columns.size(); idx++) {
-            const auto& column = row_group.columns[idx];
-            // TODO Not support for non-scalar types now.
-            const std::string format_column_name =
-                    Utils::format_name(column.meta_data.path_in_schema[0], _case_sensitive);
-            if (format_column_name == format_slot_name) {
-                // Put SlotDesc's origin column name here!
-                column_name_2_pos_in_meta.emplace(slot->col_name(), idx);
-                break;
-            }
+        const ParquetField* field = nullptr;
+        if (slot->col_unique_id() != -1) {
+            field = schema.get_stored_column_by_field_id(slot->col_unique_id());
```

Comment:
This logic has already covered what `IcebergMetaHelper` does. Could you refactor Iceberg tables too so we can remove `IcebergMetaHelper`.

Reply:
This could affect compatibility. would do this later